File: apps/tv/modules/valence-focus-lift/android/src/main/java/app/valence/tv/focuslift/ValenceFocusLiftModule.kt
package app.valence.tv.focuslift

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/** Offers JavaScript a view that lifts what it holds as the remote lands on anything inside it. */
class ValenceFocusLiftModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ValenceFocusLift")

    View(ValenceFocusLiftView::class) {
      Prop("scale") { view: ValenceFocusLiftView, scale: Float -> view.scale = scale }
      Prop("shadowHeight") { view: ValenceFocusLiftView, height: Float -> view.shadowHeight = height }
      Prop("cornerRadius") { view: ValenceFocusLiftView, radius: Float -> view.cornerRadius = radius }
      Prop("isAnchoredLeft") { view: ValenceFocusLiftView, isAnchoredLeft: Boolean ->
        view.isAnchoredLeft = isAnchoredLeft
      }
    }
  }
}

File: apps/tv/modules/valence-focus-lift/ios/ValenceFocusLiftModule.swift
import ExpoModulesCore

/// Offers JavaScript a view that lifts what it holds as the remote lands on it.
public class ValenceFocusLiftModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ValenceFocusLift")

    View(ValenceFocusLiftView.self) {
      Prop("scale") { (view: ValenceFocusLiftView, scale: Double) in
        view.scale = CGFloat(scale)
      }

      Prop("shadowHeight") { (view: ValenceFocusLiftView, height: Double) in
        view.shadowHeight = CGFloat(height)
      }

      Prop("isAnchoredLeft") { (view: ValenceFocusLiftView, isAnchoredLeft: Bool) in
        view.isAnchoredLeft = isAnchoredLeft
      }

      Prop("cornerRadius") { (view: ValenceFocusLiftView, cornerRadius: Double) in
        view.cornerRadius = CGFloat(cornerRadius)
      }
    }
  }
}
